addons/project/models/project_task.py

```diff
@@ -1742,6 +1742,12 @@ def action_convert_to_subtask(self):
             }
         }
 
+    def action_archive(self):
+        for task in self:
+            if not task.display_in_project and task.parent_id:
+                task.display_in_project = True
```

Comment:
We can write it in one line. 🙂 
```suggestion
        self.filtered(lambda t: not t.display_in_project and task.parent_id).display_in_project = True
```